src/superphot_plus/lightcurve.py

```diff
@@ -55,6 +55,43 @@ def obs_count(self, band=None):
         else:
             return np.count_nonzero(self.bands == band)
 
+    def find_max_flux(self, error_coeff=-1.0, band=None):
+        """Find the value and timestamp of the maximum flux in
+        a given band, accounting for the error estimate.
+
+        Uses value[t] = fluxes[t] + error_coeff * | flux_errors[t] |
+
+        Parameters
+        ----------
+        error_coeff : float
+            The multiplicative constant to use when accounting for
+            error. Default = -1.0 for flux - |error|.
+        band : str, optional
+            The band to check.  Use None to find the maximum over
+            all bands.
+
+        Returns
+        -------
+        value, timestamp : float, float
+            Returns the maximum flux value and its corresponding time.
+
+        Raises
+        ------
+        ValueError if the light curve is empty or there are no observations in the given band.
+        """
+        if band is None:
+            ref_band = [True] * len(self.bands)
+        else:
+            ref_band = self.bands == band
+        if np.all(ref_band == False) or len(self.bands) == 0:
+            raise ValueError(f"ERROR: Light curve has no points in band {band}")
```

Comment:
This will read "Error: Light curve has no points in " when the band is None.

Reply:
I reformatted to make it more clear.